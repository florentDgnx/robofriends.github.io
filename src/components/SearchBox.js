import React, { Component } from "react";

class SearchBox extends Component {
  render() {
    return (
      <div id="searchBox" className="pa2">
        <input
          id="searchInput"
          aria-label="Search Robot"
          className={"pa3 ba b--green bg-lightest-blue"}
          type="search"
          placeholder="search robot"
          value={this.props.searchField}
          onChange={this.props.changeInput}
        />
      </div>
    );
  }
}

export default SearchBox;
