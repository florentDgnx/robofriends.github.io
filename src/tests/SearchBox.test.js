import { shallow } from "enzyme";
import React from "react";
import SearchBox from "../components/SearchBox";

describe("test searchBox component", () => {
  it("expect to render SearchBox component", () => {
    expect(
      shallow(<SearchBox value={"mock value"} onChange={() => {}} />)
    ).toMatchSnapshot();
  });

  // it("expect change input", () => {
  //   const wrapper = shallow(<SearchBox searchField="" />);
  //   const input = wrapper.find('[id="searchInput"]');
  //   input.simulate("keypress", { key: ["t"] });
  //   console.log(input.props());
  //   expect(input.props().value).toEqual("t");
  // });
});
