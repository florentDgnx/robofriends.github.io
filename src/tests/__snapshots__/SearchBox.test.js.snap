// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test searchBox component expect to render SearchBox component 1`] = `ShallowWrapper {}`;
